content/docs/tasks/telemetry/export-logs-through-stackdriver.md

```diff
@@ -0,0 +1,259 @@
+This specifies all the variables that are available for analysis. It
+contains information like source service, destination service, auth
+metrics(coming..) among others. Following is a diagram of the pipeline:
+
+{{< image width="75%" ratio="69.52%"
+link="../img/istio-analytics-using-stackdriver.png" alt="Image is not available"
+title="Diagram of exporting logs from Istio to StackDriver for analysis" >}}
+
+Istio supports exporting logs to Stackdriver which can be extended to export
+logs to your favorite sink like BigQuery, Pub/Sub or GCS. Please follow steps
+below to setup your favorite sink for exporting logs first and then Stackdriver
+in Istio.
+
+### Setting up various log sinks
+
+Common setup for all sinks:
+
+1. Enable StackDriver Monitoring API for the project.
+1. Make sure `principalEmail` that would be setting up the sink has write access to the project and Logging Admin role permissions.
+
+#### BigQuery
+
+1.  Create a BigQuery dataset as a destination for the logs export.
+1.  Record the ID of the dataset. It will be needed to configure the Stackdriver handler.
+    It would be of the form `bigquery.googleapis.com/projects/[PROJECT_ID]/datasets/[DATASET_ID]`
+1.  Give [sink’s writer identity](https://cloud.google.com/logging/docs/api/tasks/exporting-logs#writing_to_the_destination):     `cloud-logs@system.gserviceaccount.com` BigQuery Data Editor role in IAM.
+
+#### Google Cloud Storage (GCS)
+
+1.  Create a GCS bucket where you would like logs to get exported in GCS.
+1.  Recode the ID of the bucket. It will be needed to configure Stackdriver.
+    It would be of the form `storage.googleapis.com/[BUCKET_ID]`
+1.  Give [sink’s writer identity](https://cloud.google.com/logging/docs/api/tasks/exporting-logs#writing_to_the_destination):     `cloud-logs@system.gserviceaccount.com` Storage Object Creator role in IAM.
+
+#### Google Cloud Pub/Sub
+
+1.  Create a topic where you would like logs to get exported in Google Cloud Pub/Sub.
+1.  Recode the ID of the topic. It will be needed to configure Stackdriver.
+    It would be of the form `pubsub.googleapis.com/projects/[PROJECT_ID]/topics/[TOPIC_ID]`
+1.  Give [sink’s writer identity](https://cloud.google.com/logging/docs/api/tasks/exporting-logs#writing_to_the_destination):     `cloud-logs@system.gserviceaccount.com` Pub/Sub Publisher role in IAM.
+
+### Setting up Stackdriver
+
+A Stackdriver handler must be created to export data to Stackdriver. The configuration schema
+for a Stackdriver handler can be found [here](https://github.com/istio/api/blob/0142cbd81f9dc723c57c1d43b1e07d3cb0817253/policy/v1beta1/cfg.proto#L243:9).
+Config proto for Stackdriver can be found
+[here](https://github.com/istio/istio/blob/master/mixer/adapter/stackdriver/config/config.proto).
+Handler is configured based on this proto.
+
+1.  Save following yaml file as `stackdriver.yaml`. Replace `<project_id>,
```

Comment:
Save following yaml file -> Save the following

Reply:
Fixed